### PR TITLE
Listen on foundry & truffle client

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

**Review by bunsenstraat — COMMENTED** [review]
any tests for this?

**Comment by yann300** [comment]
yes I 'll add that pretty soon

File: libs/remixd/src/services/foundryClient.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  listenOnFoundryCompilation () {
+    try {
+      const buildPath = utils.absolutePath('out', this.currentSharedFolder)
+      const watcher = chokidar.watch(buildPath, { depth: 0, ignorePermissionErrors: true })
+      const compilationResults = {}
+      watcher.on('change', async (f: string) => {
+        const content = await fs.readFile(f, { encoding: 'utf-8' })
+        const compilationResult = JSON.parse(content)
+        if (f.endsWith('metadata.json')) {
```
Comment (ioedeveloper) [inline libs/remixd/src/services/foundryClient.ts:62]: Empty block/logic

File: libs/remixd/src/services/foundryClient.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      })
+    })
+  }
+
+  listenOnFoundryCompilation () {
+    try {
+      const buildPath = utils.absolutePath('out', this.currentSharedFolder)
+      const watcher = chokidar.watch(buildPath, { depth: 0, ignorePermissionErrors: true })
+      const compilationResults = {}
+      watcher.on('change', async (f: string) => {
```
Comment (ioedeveloper) [inline libs/remixd/src/services/foundryClient.ts:59]: No error handler for change listener callback.

Comment (yann300) [reply libs/remixd/src/services/foundryClient.ts:67]: there isn't any error handler for this event

**Comment by yann300** [comment]
@bunsenstraat i've added test

**Review by bunsenstraat — APPROVED** [review]

**Pull request merged**